pycparser/c_ast.py

```diff
@@ -84,12 +87,15 @@ def show(self, buf=sys.stdout, offset=0, attrnames=False, nodenames=False, showc
             buf.write(lead + self.__class__.__name__+ ': ')
 
         if self.attr_names:
+            nvlist = []
```

Comment:
If you factor out the emptiness check into a helper function, this loop can become a simple loop comprehension again:

```
nvlist = [(n, getattr(self, n)) for n in self.attr_names if ...]
```

Or something similar